src/app/new-auth/register/components/recaptcha/recaptcha.directive.ts

```diff
@@ -0,0 +1,193 @@
+    private element: ElementRef,
+    private ngZone: NgZone,
+    private injector: Injector,
+    private validator: RecaptchaValidator,
+
+  ) {}
+
+  ngOnInit() {
+    this.registerReCaptchaCallback();
+    this.addScript();
+  }
+
+  registerReCaptchaCallback() {
+    window.reCaptchaLoad = () => {
+      const config = {
+        ...this.config,
+        sitekey: this.key,
+        callback: this.onSuccess.bind(this),
+        'expired-callback': this.onExpired.bind(this)
+      };
+      this.widgetId = this.render(this.element.nativeElement, config);
+    };
+  }
+
+  ngAfterViewInit() {
+    this.control = this.injector.get(FormControl);
+    this.setValidators();
+  }
+
+  /**
+   * Useful for multiple captcha
+   *
+   * @returns {number}
+   */
+  getId() {
+    return this.widgetId;
+  }
+
+  /**
+   * Calling the setValidators doesn't trigger any update or value change event.
+   * Therefore, we need to call updateValueAndValidity to trigger the update
+   */
+  private setValidators() {
+    this.control.setValidators(Validators.required);
+    this.control.updateValueAndValidity();
+  }
+
+  writeValue(obj: any): void {}
+
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  /**
+   * onExpired
+   */
+  onExpired() {
+    this.ngZone.run(() => {
+      this.captchaExpired.emit();
+      this.onChange(null);
+      this.onTouched(null);
+      this.captchaSuccess.emit(false);
+    });
+  }
+
+  /**
+   *
+   * @param response
+   */
+  onSuccess(token: string) {
+    this.ngZone.run(() => {
+      this.verifyToken(token);
+      this.captchaResponse.next(token);
+      this.onChange(token);
+      this.onTouched(token);
+      this.captchaSuccess.emit(true);
+    });
+  }
+
+  /**
+   *
+   * @param token
+   */
+  verifyToken(token: string) {
+    this.validator.validateToken(token);
+  }
+
+  /**
+   * Renders the container as a reCAPTCHA widget and returns the ID of the newly created widget.
+   *
+   * @param element
+   * @param config
+   * @returns {number}
+   */
+  private render(element: HTMLElement, config): number {
+    return grecaptcha.render(element, config);
+  }
+
+  /**
+   * Resets the reCAPTCHA widget.
+   */
+  reset(): void {
+    if (!this.widgetId) {
+      return;
+    }
+    grecaptcha.reset(this.widgetId);
+    this.onChange(null);
+  }
+
+  /**
+   * Gets the response for the reCAPTCHA widget.
+   *
+   * @returns {string}
+   */
+  getResponse(): string {
+    if (!this.widgetId) {
+      return grecaptcha.getResponse(this.widgetId);
+    }
+  }
+
+  /**
+   * Add the script
+   */
+  addScript() {
+    const script = document.createElement('script');
+    const lang = this.lang ? '&hl=' + this.lang : '';
+    script.src = `https://www.google.com/recaptcha/api.js?onload=reCaptchaLoad&render=explicit${lang}`;
+    script.async = true;
+    script.defer = true;
+    document.body.appendChild(script);
+  }
```

Comment:
This file needs better block comments in general.  This particular function can also be more descriptive.